src/http/contract-registry/ContractRegistry.test.ts

```diff
@@ -0,0 +1,335 @@
+import { IContract, IRegisterContractRequest } from '../types'
```

Comment:
Did you write this file entirely from scratch? If not, and you just refactor pieces from original grpc one, it would be good to keep history by moving the file and then making changes on top

Reply:
the test file ? Yes I wrote it from scratch